enhancements/update/update-status-api.md

```diff
@@ -0,0 +1,398 @@
+
+This proposal intends to capture only a limited part of the roadmap for a system reporting update health and status. The [Update Health API Controller Proposal][update-health-api-dd] document describes a potential future system where individual (and possibly external) domain-responsible components would produce update insights into the `UpdateStatus` API. This enhancement proposal corresponds to only Stage 1 of the implementation roadmap outlined there. There has yet to be a consensus on further stages, and this enhancement does not aim to achieve it. Any hypothetical modular system will be a much larger effort to implement, and its design will benefit from experience and feedback obtained by seeing API consumed by users, backed by the proposed simple implementation in the USC.
+
+### Workflow Description
+
+1. On a cluster where the Status API is enabled, the CVO manages a new Update Status Controller (USC) deployment as its operand.
+1. The USC monitors the cluster and maintains an `UpdateStatus` resource called `cluster`.
+1. While there is no update happening, respective conditions convey:
+   1. `.status.controlPlane.conditions` has a `Updating=False` condition.
+   1. `.status.workerPools[*].conditions` each have a `Updating=False` condition.
+1. When a user runs `oc adm upgrade status`, the `oc` client reads the `UpdateStatus` and reports that no update is happening.
+1. The user triggers the update.
+1. The USC monitors `ClusterVersion`, `ClusterOperator`, `MachineConfigPool`, and `Node` resources and reflects the state of the update through the `UpdateStatus` resource via a set of status and health insights.
+1. When the user runs `oc adm upgrade status`, the client reads the `UpdateStatus` and uses status insights to convey progress and health insights to convey issues the admin needs to address.
+
+### API Extensions
+
+Introduce a new Cluster-scoped `UpdateStatus` CRD to convey update status information. For now, its `.spec` is empty but can be used in the future to configure the desired form or content of the information surfaced in status.
+
+The `status` subresource content is the primary value of the enhancement. The status and health information about the update are expressed through the `status` subresource.
+
+Introduce a new `update.openshift.io` group for `UpdateStatus`. There is no suitable existing OpenShift API group for where `UpdateStatus` would fit. The `ClusterVersion` CRD, which is used to convey some information about the update process, is in the `config.openshift.io` group. `UpdateStatus` provides no configuration capabilities, so `config.openshift.io` is a poor fit. The new group is well-suited to contain some of the other update-related APIs needed for further incoming features like maintenance windows or update pre-checks.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
```

Comment:
Can we clearly express in a sentence who is the consumer of this API in the hcp context? is it any managed services powered by hypershift.openshift.io? is it the cluster admin?